### DISCUSSION TITLE
Policy has all the necessary "ha-" features but there is no mirror.

### DISCUSSION THREAD

**Discussion by brycechesternewman** [rabbitmq/rabbitmq-server#9377] [Questions]
Hello,
We are running RabbitMQ 3.11.19 and Erlang 25.3.2.2.
We will sometimes add new nodes with new AWS AMI's that contain new security updates etc.
The new nodes join the cluster successfully, the cluster topology looks normal, then we tear down the old nodes using rabbitmqctl drain, then the AWS ASG will terminate the old nodes and the rabbit_peer_discovery_aws will remove the node from the cluster.
We have a three node cluster, but have also seen this problem using a two node cluster in our production environment. We have not been able to reproduce yet in the lower environments.
Notice two queues below, the policy has all the nessarry "ha-" features but there is no mirror. A few weeks ago we noticed this after a deployment, so in order to fix this we needed to apply a new temp policy(with all the same features), then remove the temp policy. Just yesterday we added new new nodes to the cluster and the new nodes have the same version; RabbitMQ 3.11.19 and Erlang 25.3.2.2. What is really scary about this issue is 1) the queues are not being mirrored and 2) we had 15 queues disappear from the cluster during that did have mirrors before the old nodes left the cluster. 

Here are two examples. The queues are not mirrored and we left them this way to debug what might be going on. What can we look for(aka is the mnesia database corrupt?) and how can we fix or prevent this from happening?
```
Features	
durable:	true
queue storage version:	1
Policy	[dsa-low-queue-policy](https://<omitted>/#/policies/dsa/dsa-low-queue-policy)
Operator policy	
Effective policy definition	
ha-mode:	exactly
ha-params:	2
ha-sync-batch-size:	500
ha-sync-mode:	automatic
max-length:	1000000
queue-master-locator:	min-masters
queue-mode:	lazy
Node	rabbit@<omitted>
Mirrors	
State	idle
Consumers	1
Consumer capacity 	100%
```

```
Features	
durable:	true
queue storage version:	1
Policy	[dsa-policies](https://<omitted>/#/policies/dsa/dsa-policies)
Operator policy	
Effective policy definition	
ha-mode:	exactly
ha-params:	2
ha-sync-batch-size:	500
ha-sync-mode:	automatic
queue-master-locator:	min-masters
Node	rabbit@<omitted>
Mirrors
State	idle
Consumers	1
Consumer capacity 	100%
```

**Comment by michaelklishin** [comment]
Classic mirrored queues have been deprecated for a couple of years now, scheduled to be [removed in 4.0](https://blog.rabbitmq.com/posts/2021/08/4.0-deprecation-announcements/), which is expected to come out in Q1-Q2 next year. You must switch to quorum queues and/or streams.

By far the most likely reason is that [other policies that match the same set of queues have a higher priority](https://rabbitmq.com/parameters.html#policy-priorities).

**Reply by brycechesternewman** [reply]
Thank you. We will look to switching to quorum queues.

> By far the most likely reason is that [other policies that match the same set of queues have a higher priority](https://rabbitmq.com/parameters.html#policy-priorities).

Unfortunately, this is not the case.

**Reply by brycechesternewman** [reply]
btw: We think the problem starting showing up after upgrading to 3.11.19.
